apis/v1alpha2/targetallocator_types.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +kubebuilder:skip
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TargetAllocatorSpec defines the desired state of TargetAllocator.
+type TargetAllocatorSpec struct {
+	// Common defines fields that are common to all OpenTelemetry CRD workloads.
+	OpenTelemetryCommonFields `json:",inline"`
+	// AllocationStrategy determines which strategy the target allocator should use for allocation.
+	// The current options are least-weighted and consistent-hashing. The default option is consistent-hashing
+	// +optional
+	// +kubebuilder:default:=consistent-hashing
+	AllocationStrategy TargetAllocatorAllocationStrategy `json:"allocationStrategy,omitempty"`
+	// FilterStrategy determines how to filter targets before allocating them among the collectors.
+	// The only current option is relabel-config (drops targets based on prom relabel_config).
+	// Filtering is disabled by default.
+	// +optional
+	FilterStrategy string `json:"filterStrategy,omitempty"`
+	// ServiceAccount indicates the name of an existing service account to use with this instance. When set,
+	// the operator will not automatically create a ServiceAccount for the TargetAllocator.
+	// +optional
+	PrometheusCR TargetAllocatorPrometheusCR `json:"prometheusCR,omitempty"`
+}
+
+// TargetAllocatorPrometheusCR configures Prometheus CustomResource handling in the Target Allocator.
+type TargetAllocatorPrometheusCR struct {
+	// Enabled indicates whether to use a PrometheusOperator custom resources as targets or not.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+	// Interval between consecutive scrapes. Equivalent to the same setting on the Prometheus CRD.
```

Comment:
is there any precedence order for the interval defined here and on the service/pod monitors?

Reply:
It's exactly the same as the respective setting on the Prometheus CR - a default that can be overridden in individual CRs.

Reply:
is it smth that should be documented here?

Reply:
The comment already says it's equivalent to the Prometheus CR attribute, but I can also mention it's an overridable default.

Reply:
Done.